File: SW/iot-server-dreamsedge-v1.1.0/Backend/app/routes/api/server_management.py
from flask import Blueprint, request, jsonify, render_template


import asyncio

server_manage_bp = Blueprint("server_management", __name__)

@server_manage_bp.route("/server_management")
def server_management():
    return render_template("server_management.html")

# ------------------------------------------------------
# Check server status function: 
# -> Check status of TCP server and AWS server
# ------------------------------------------------------
@server_manage_bp.route("/check_status_server", methods=["GET"])
def check_status_server():
    return jsonify({
        "tcp_status": "running",
        "aws_status": "connected"
    })

File: SW/Test/iot-server-dreamsedge-v1.1.0/Frontend/static/js/server_management.js
$(document).ready(function() {
    function fetchServerStatus() {
        $.get("/check_status_server", function (response) {
            // Update the TCP Server status UI
            updateStatus("tcpServerStatus", "tcpStatusText", response.tcp_status, {
                "running": "green",
                "stopping": "red",
                "timeout": "red"
            });

            // Update the AWS Server status UI
            updateStatus("awsServerStatus", "awsStatusText", response.aws_status, {
                "connected": "green",
                "disconnected": "red",
                "timeout": "red"
            });
        }).fail(function () {
            $("#tcpStatusText, #awsStatusText").text("Error");
        });
    }

    function fetchDeviceManagement() {
        $.get("/get_device_management", function (response) {
            var tableBody = $("#deviceManageTable tbody");
            tableBody.empty();
            
            response.devices.forEach(function(device) {
                var row = `<tr><td>${device.id}</td><td>${device.last_message}</td></tr>`;
                tableBody.append(row);
            });
        }).fail(function () {
            console.log("Failed to fetch device management data");
        });
    }

    function updateStatus(cardId, textId, status, statusColors) {
        $(`#${textId}`).text(status.charAt(0).toUpperCase() + status.slice(1)); 
        let colorClass = statusColors[status] || "red";
        $(`#${cardId}`).removeClass("red green").addClass(colorClass);
    }

    // Call the functions when the page loads
    fetchServerStatus();
    fetchDeviceManagement();
});
